File: backend/models/daily_log.go
package models

import "time"

type DailyLog struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	Date      time.Time `json:"date"`
	Mood      *int      `json:"mood,omitempty"`
	Energy    *int      `json:"energy,omitempty"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type DailyLogRequest struct {
	Date   string `json:"date" binding:"required"`
	Mood   *int   `json:"mood,omitempty"`
	Energy *int   `json:"energy,omitempty"`
}

type EnergyCorrelationPoint struct {
	EnergyLevel    int     `json:"energy_level"`
	CompletionRate float64 `json:"completion_rate"`
}

File: mobile/lib/features/checkin/models/daily_log.dart
class DailyLogData {
  final String? id;
  final int? mood;
  final int? energy;

  const DailyLogData({this.id, this.mood, this.energy});

  factory DailyLogData.fromJson(Map<String, dynamic> json) {
    return DailyLogData(
      id: json['id'] as String?,
      mood: json['mood'] as int?,
      energy: json['energy'] as int?,
    );
  }
}
